var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -83,7 +83,14 @@ def setup_dependent_package(self, module, dep_spec):
         else:
             raise RuntimeError('No suitable bindir found')
 
-        self.spec.mpicc = join_path(bindir, 'mpicc')
```

Comment:
We went through it multiple times. The current code is perfectly valid even when intel compilers are used